Automattic/jetpack
# Blocks: do not display block placeholder to non-admins

## Proposed changes:

We currently display a placeholder in the block editor when a block cannot be used because its associated module is not active. The placeholder offers a button inviting folks to activate the module.

Since only admins can activate modules, the placeholder is useless to non-admins. Let's consequently skip registering the blocks for non-admins when the associated module is not active.

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Start with a site that's connected to WordPress.com.
* Create an additional user with an Author or Editor role.
* In Jetpack > Settings > Modules, deactivate all modules but blocks.
* Log in to the site as the author / editor
* Go to Posts > Add New
* Try to insert a Related Posts block by typing `/related`
    * The block should not be available to be inserted.
    * You can repeat this with the other blocks touched by this PR:
       * Blog Stats
       * Top Posts
       * Ad block
       * Contact Form
       * VideoPress
* Open the Jetpack sidebar
    * You should not see the SEO, Sharing & Likes, or Publicize panels.
* Now log back in as the admin.
* Go to Posts > Add New
* Try to insert the blocks I mentioned above.
    * You should see all blocks as available.
    * Use the buttons in the placeholders to turn the feature on.
* Now log back in as the author / editor.
* Go to Posts > Add New
* Try to insert the blocks again
    * They should now be available to you.
